core/quantity/src/Formatter/Formatter.ts

```diff
@@ -214,7 +214,26 @@ export class Formatter {
       if (i > 0 && unitConversion.offset !== 0) // offset should only ever be defined for major unit
         throw new QuantityError(QuantityStatus.InvalidCompositeFormat, `The Format ${spec.format.name} has a invalid unit specification..`);
 
-      let unitValue = applyConversion(posMagnitude, unitConversion) + this.FPV_MINTHRESHOLD;
+      // let unitValue = applyConversion(posMagnitude, unitConversion) + this.FPV_MINTHRESHOLD;
+      // put this in try catch, if catched QuantityError(QuantityStatus.InvertingZero), if the spec.format.type is Ratio, we will return 1:0
+
+      let unitValue = 0.0;
+      try {
+        unitValue = applyConversion(posMagnitude, unitConversion) + this.FPV_MINTHRESHOLD;
+      } catch (e) {
+        // error code 35049 is thrown when inverting zero
+        // This is special case for ratio, just return 1:0
+        if (spec.format.type === FormatType.Ratio && e instanceof QuantityError && e.errorNumber === QuantityStatus.InvertingZero) {
```

Comment:
Oh, I didn't think of this when I created the applyConversion function. I don't like that we are using an exception here for something that's expected. All the other spots that dealt with conversion before which I adjusted to use this method didn't support that conversion could fail. So I thought raising an exception is the best choice, so we can be sure it's not swallowed. But looking at this here.... In native we return a UnitsProblemCode which contains these errors:
```cpp
enum class UnitsProblemCode
    {
    NoProblem = 0,
    UncomparableUnits = 31,     //!< Units provided on the argument list are not comparable
    InvalidUnitName = 32,       //!< Not-recognizd unit name or unit is not associated with a Phenomenon
    InvertingZero = 33,         //!< Attempting to invert a zero value
    UnspecifiedProblem = 34,    //!< A problem occurred but the cause is not specified
    NaN = 35,                   //!< A value is NaN
    };
```

Let me think about this for a bit. I'm leaning towards a tryConvert() function which will return errors instead of throwing exceptions

Reply:
Alright, I thought about it and decided to keep it as is. Most of those problemCodes would not apply here, and creating a separate API just for this corner case seems overkill.